storage/src/meta/mod.rs

```diff
@@ -452,14 +471,16 @@ impl BlobMetaInfo {
                 index += 1;
                 let entry = &infos[index];
                 self.validate_chunk(entry)?;
-                if entry.uncompressed_offset() != last_end {
-                    return Err(einval!(format!(
-                        "mismatch uncompressed {} size {} last_end {}",
-                        entry.uncompressed_offset(),
-                        entry.uncompressed_size(),
-                        last_end
-                    )));
-                }
+
+                // FIXME: for stargz chunks, disable this check.
+                // if entry.uncompressed_offset() != last_end {
```

Comment:
How about allocating a flags in the `meta_ci_flags` field for stargz so we could special handling stargz blobs?

Reply:
Fixed